convert2rhel/pkghandler.py

```diff
@@ -608,6 +609,27 @@ def remove_non_rhel_kernels():
     return non_rhel_kernels
 
 
+def fix_default_kernel():
+    """
+        On systems converted from OL 6 and latter, DEFAULTKERNEL may contain kernel-uek after the conversion.
+
+        The solution handled by this function is to remove the  kernel-uek from DEFAULTKERNEL setting the correct
+        kernel according with the rhel version.
+        """
+    if "oracle" not in system_info.name.lower() or system_info.arch == "s390x":
+        # Applicable only on systems derived from Oracle Linux, and systems using GRUB2 (s390x uses zipl)
+        return
+
+    loggerinst = logging.getLogger(__name__)
+
+    loggerinst.info("Checking remaining oracle linux kernel and fix to RHEL kernel...")
+    kernel_syscfg = utils.get_file_content("/etc/sysconfig/kernel")
+    # need to change to "kernel" in rhel6, 7 and "kernel-core" in rhel8
+    new_kernel_str = "DEFAULTKERNEL=" + ("kernel" if int(system_info.version) in [6, 7] else "kernel-core")
+    kernel_syscfg = kernel_syscfg.replace("DEFAULTKERNEL=kernel-uek", new_kernel_str)
```

Comment:
We probably also want to handle replacing the CentOS Plus kernel (`kernel-plus`) too...

Reply:
Hi @Conan-Kudo !  Thanks for the suggestion, changed!